Assertion violation happens here.
Counterexample:
a = 0, y = (- 1)
tmp = 0

Transaction trace:
B8.constructor()
State: a = 0, y = 0
B8.f8()
    a.get8() -- untrusted external call
  --> origin.sol:28:3:
   |
28 |            assert(y > 0);
   |            ^^^^^^^^^^^^^
